### DISCUSSION TITLE
Bulk Actions to show number of selected items

### DISCUSSION THREAD

**Discussion by fakeheal** [Power-Components/livewire-powergrid#800] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary

It'd be really cool if you can 'track' how many items you've checked in a table across all pages when using the bulk action feature.

### Why is this needed?

It helps keep track of the number of items selected before performing a bulk action.

### Suggested Solution(s)

A way to track the count of `checkboxValues`.

I tried doing the following in my Livewire Powergrid table:

```php
private Button $deleteBulkAction;

public function __construct($id = nul) {
    parent::__construct($id);
    $this->deleteBulkAction = Button::add('bulk-delete')
        ->caption('Bulk Delete (' . count($this->checkedValues()) . ')')
        ->class('cursor-pointer bg-red-500 ' . self::HEADER_ACTION_BASE_CLASS)
        ->emit('bulkDelete', []);
}

public function updatedCheckboxValues(): void
{
    $this->deleteBulkAction->caption = 'Bulk Delete (' . count($this->checkedValues()) . ')';
}
```

But it doesn't work properly. Maybe be I'm misunderstanding some core livewire concept. If that's the case I'd like to apologize in advance.

### Related issue(s)/PR(s)

None so far.

**Comment by luanfreitasdev** [comment]
This is a great addition, it will be released in version 4

**Reply by luanfreitasdev** [reply]
_"But it doesn't work properly. Maybe be I'm misunderstanding some core livewire concept. If that's the case I'd like to apologize in advance."_

Unfortunately this way will not work until some request because of wire:model defer

```html
wire:model.defer="checkboxValues"
```

**Comment by CikaDraza** [comment]
How to get IDs of table rows to delete in bulk?
Here is how I get IDs from PowerGrid documentation.
`    public function header(): array
    {
        return [
            Button::add('bulk-delete')
                ->slot('Bulk delete (<span x-text="window.pgBulkActions.count(\'' . $this->tableName . '\')"></span>)')
                ->class('pg-btn-white dark:ring-pg-primary-600 dark:border-pg-primary-600 dark:hover:bg-pg-primary-700 dark:ring-offset-pg-primary-800 dark:text-pg-primary-300 dark:bg-pg-primary-700')
                ->dispatch('bulkDelete.' . $this->tableName, []),
        ];
    }

    #[On('bulkDelete.{tableName}')]
    public function bulkDelete(): void
    {
        $this->js('alert(window.pgBulkActions.get(\'' . $this->tableName . '\'))');
    }`

But I need to pass IDs to the modal when it opens.
`    public function header(): array
    {
        return [
            Button::add('bulk-delete')
                ->slot('Bulk delete (<span x-text="window.pgBulkActions.count(\'' . $this->tableName . '\')"></span>)')
                ->class('pg-btn-white dark:ring-pg-primary-600 dark:border-pg-primary-600 dark:hover:bg-pg-primary-700 dark:ring-offset-pg-primary-800 dark:text-pg-primary-300 dark:bg-pg-primary-700')
                ->openModal('form.confirm-delete-appointments', [], 'js', [
                    'selectedIds' => "window.pgBulkActions.get('{$this->tableName}')",
                ]),
        ];
    }`
